mailpoet/assets/js/src/public.jsx

```diff
@@ -77,6 +77,76 @@ jQuery(($) => {
     });
   }
 
+  function submitSubscribeForm(form, formData, parsley) {
+    form.addClass('mailpoet_form_sending');
+    // ajax request
+    MailPoet.Ajax.post({
+      url: window.MailPoetForm.ajax_url,
+      token: formData.token,
+      api_version: formData.api_version,
+      endpoint: 'subscribers',
+      action: 'subscribe',
+      data: formData.data,
+    })
+      .fail((response) => {
+        if (
+          response.meta !== undefined &&
+          response.meta.redirect_url !== undefined
+        ) {
+          // go to page
+          window.top.location.href = response.meta.redirect_url;
+        } else {
+          if (response.meta && response.meta.refresh_captcha) {
+            updateCaptcha();
+          }
+          form
+            .find('.mailpoet_validate_error')
+            .html(response.errors.map((error) => error.message).join('<br />'))
+            .show();
+        }
+      })
+      .done((response) => {
+        if (window.grecaptcha && formData.recaptcha) {
+          window.grecaptcha.reset(formData.recaptcha);
+        }
+        return response;
+      })
+      .done((response) => {
+        // successfully subscribed
+        if (
+          response.meta !== undefined &&
+          response.meta.redirect_url !== undefined
+        ) {
+          setFormCookieAfterSubscription(form);
+          // go to page
+          window.location.href = response.meta.redirect_url;
+        } else {
+          displaySuccessMessage(form);
+        }
+
+        // reset form
+        form.trigger('reset');
+        // reset validation
+        parsley.reset();
+        // reset captcha
+        if (window.grecaptcha && formData.recaptcha) {
+          window.grecaptcha.reset(formData.recaptcha);
+        }
```

Comment:
I also noticed that we are calling window.grecaptcha.reset() twice and I'm not sure why. I opted to leave the code as is since this duplication was already present and, when creating this PR, I did not have time to investigate the reasons.